src/sinks/datadog/mod.rs

```diff
@@ -36,13 +36,13 @@ pub(crate) fn default_site() -> String {
 pub struct DatadogCommonConfig {
     /// The endpoint to send observability data to.
     ///
-    /// The endpoint must contain an HTTP scheme, and may specify a
-    /// hostname or IP address and port.
+    /// The endpoint must contain an HTTP scheme, and may specify a hostname or IP
+    /// address and port. The path must be specified if expected by the downstream.
     ///
     /// If set, overrides the `site` option.
     #[configurable(metadata(docs::advanced))]
-    #[configurable(metadata(docs::examples = "http://127.0.0.1:8080"))]
-    #[configurable(metadata(docs::examples = "http://example.com:12345"))]
+    #[configurable(metadata(docs::examples = "http://127.0.0.1:8080/api/v2/logs"))]
+    #[configurable(metadata(docs::examples = "http://example.com:12345/api/beta/sketches"))]
```

Comment:
These examples are a bit weird when seen across different `datadog_*` sinks. For example, seeing an example of `http://example.com:12345/api/beta/sketches` for the `datadog_logs` sink is a little confusing.

I'm also wondering how this option is used by the `datadog_metrics` sink which has two different paths it needs to send to.

Two possible ideas:

* Modify the behavior of `endpoint` to actually append the path in each of the sinks
* Move this option out of `DatadogCommonConfig` and into each of the `datadog_*` sinks so that the description and examples can differ.

Reply:
> I'm also wondering how this option is used by the datadog_metrics sink which has two different paths it needs to send to.

Ahhh you know what- you just solved the mystery that was bothering me. I've used this option before with the metrics sink and didn't recall having to add the path.
That's because the metrics sink is appending it.

The use case I was in that led me to open this issue was working with the logs sink. 

Thus, I think

> Modify the behavior of endpoint to actually append the path in each of the sinks

That option would be the best one.

Reply:
Aha, yeah, I agree. It would be a breaking change, but should be a minor one since users shouldn't, typically, be using this config value so I think we can just make it and document it in the notes for the next release.